ProcessMaker/processmaker
# Resolve Task Progression and Error in Guided Templates

This PR addresses an ongoing bug in CI servers related to task progression and error handling in Guided Templates. The issues included a 422 error indicating the task was already closed, replication of closed tasks displaying, and incorrect task statuses (In Progress instead of Completed) upon completion.

## Cause

1. The <task> component was not automatically progressing to the next task.
2. An error was thrown from Screen Builder upon closing the first task, with an unknown `getScreenEndpoint` message.

## Solution

1. Removed the initial `getNextTask` method and allow the <task> component progresses correctly.
2. Updated the conditional to check if the `getScreenEndpoint` variable exists, preventing an error and allowing the <task> component to progress.

## How to Test

1. Install the latest Send Email Connector.
2. Symlink the latest Screen Builder.
3. Run `php artisan processmaker:sync-guided-templates`.
4. Navigate to Processes > Guided Templates.
5. Select a guided template.
6. Click 'Start Now' and complete the process (use 'Skip' if needed).
7. Verify no 422 errors occur.
8. Confirm that previously closed tasks do not replicate.
9. After completing the process, check Requests to ensure it has a status of 'Completed'.

## Related Tickets & Packages
- Screen Builder PR - [https://github.com/ProcessMaker/screen-builder/pull/1514]( https://github.com/ProcessMaker/screen-builder/pull/1514)
- [https://processmaker.atlassian.net/browse/FOUR-13331](https://processmaker.atlassian.net/browse/FOUR-13331)

ci:next
ci:screen-builder:observation/FOUR-13331

## Code Review Checklist
